### PR TITLE
Whats changed

### PR DESCRIPTION
**Pull request by railsagainstignorance**
after every update, articles from the week *before* the current in-memory set are searched for entities, and any in-memory entities which do not crop up in that week before are considered 'new' and available as an endpoint /newlyAppearedEntities

### CODE REVIEW

File: bin/lib/correlate.js [hunk 1]
```diff
@@ -288,24 +290,72 @@ function calcIslandSortedByCount(island){
...
+	// hence newEntities
+
+	const  afterSecs = earliestAfterSecs - (60 * 60 * 24 * 7); // i.e. one week before prev oldest
+	const beforeSecs = earliestAfterSecs;
+
+	return getLatestEntitiesMentioned(afterSecs, beforeSecs)
+		.then( deltaEntities => {
+			const newEntitiesWithCount = Object.keys(knownEntities)
+			.filter(entity => {
+				return ! deltaEntities.hasOwnProperty(entity);
```
Comment (seanmtracey) [inline bin/lib/correlate.js]: ` ! ` with the spaces... I've never seen that before :shudder:

File: bin/lib/correlate.js [hunk 2]
```diff
@@ -288,24 +290,72 @@ function calcIslandSortedByCount(island){
...
+			})
+			.map( entity => {
+				return [entity, knownEntities[entity] ];
+			});
+			newEntitiesWithCount.sort( (a,b) => {
+				if     ( a[1] > b[1] ) { return -1; }
+				else if( a[1] < b[1] ) { return +1; }
```
Comment (seanmtracey) [inline bin/lib/correlate.js]: Why `+1` and not 1?

File: bin/lib/correlate.js [hunk 3]
```diff
@@ -314,14 +364,28 @@ function fetchUpdateCorrelations(afterSecs, beforeSecs) {
...
+			endUpdatesMillis = Date.now();
+			return newCounts;
+		} )
+		.then( newCounts => {
+			startNewlyAppearedMillis = Date.now();
+			return fetchNewlyAppearedEntities()
+			.then( newEntities => {
+				endNewlyAppearedMillis = Date.now();
+				return newCounts;
+			})
```
Comment (seanmtracey) [inline bin/lib/correlate.js]: Are you `catching` the errors further up the chain?

Comment (railsagainstignorance) [inline bin/lib/correlate.js]: some catching goes on, but it is not comprehensive

**Pull request merged**